.github/workflows/tests.yml

```
@@ -0,0 +1,87 @@
+name: "Tests"
+
+on:
+  - pull_request
+  - push
+
+jobs:
+  test:
+    name: PHP:${{ matrix.php }} PHP_INVOKER:${{ toJSON(matrix.php_invoker) }} ALLOW_FAILURES:${{ toJSON(matrix.experimental) }} COVERAGE:${{ toJSON(matrix.coverage) }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        experimental: [false]
+        coverage: [false]
+        php:
+          - "5.3"
+          - "5.4"
+          - "5.5"
+          - "5.6"
+          - "7.0"
+          - "7.1"
+          - "7.2"
+          - "7.3"
+        php_invoker:
+          - true
+          - false
+        include:
+          - php: "7.4"
+            php_invoker: true
+            coverage: true
+          - php: "7.4"
+            php_invoker: false
+            coverage: true
+          - php: "8.0"
+            php_invoker: true
+            experimental: false
+            coverage: false
+          - php: "8.0"
+            php_invoker: false
+            coverage: false
+            experimental: false
+          - php: "8.1"
+            php_invoker: true
+            experimental: true
+            coverage: false
+          - php: "8.1"
+            php_invoker: false
+            experimental: true
+            coverage: false
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      - name: Setup PHP ${{ matrix.php }}
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          ini-values: display_errors=off, log_errors=on
+          extensions: xdebug, ctype, dom, json, pcre, reflection, spl
+        env:
+          # https://github.com/shivammathur/setup-php/issues/407#issuecomment-773675741
+          fail-fast: true
+
+      - name: Switch composer to v1
+        # Since "The PEAR repository has been removed from Composer 2.0" it makes "composer install" to fail on this repo
+        # revert to composer v1 (and run it outside the checkouted repo, because composer v2 is basically stuck on that error on any command)
+        run: |
+          composer self-update --1
+
```

would it be enough to already specify that we want to use composer v1 like this, instead of reverting it from v2 to v1? (it was necessary on travis but probably it is as simple as this on gha)
```suggestion
        with:
          php-version: ${{ matrix.php }}
          ini-values: display_errors=off, log_errors=on
          extensions: xdebug, ctype, dom, json, pcre, reflection, spl
          # Use composer v1 because it does not work with v2: "The PEAR repository has been removed from Composer 2.0"
          tools: composer:v1
        env:
          # https://github.com/shivammathur/setup-php/issues/407#issuecomment-773675741
          fail-fast: true

```

---

it's awesome! thanks!